pkg/p2p/kadcast/writer/broadcast.go

```diff
@@ -0,0 +1,117 @@
+// This Source Code Form is subject to the terms of the MIT License.
+// If a copy of the MIT License was not distributed with this
+// file, you can obtain one at https://opensource.org/licenses/MIT.
+//
+// Copyright (c) DUSK NETWORK. All rights reserved.
+
+package writer
+
+import (
+	"bytes"
+	"context"
+	"errors"
+
+	"github.com/dusk-network/dusk-blockchain/pkg/config"
+	"github.com/dusk-network/dusk-blockchain/pkg/p2p/wire/protocol"
+	"github.com/dusk-network/dusk-blockchain/pkg/p2p/wire/topics"
+	"github.com/dusk-network/dusk-blockchain/pkg/util/container/ring"
+	"github.com/dusk-network/dusk-blockchain/pkg/util/nativeutils/eventbus"
+	"github.com/dusk-network/dusk-protobuf/autogen/go/rusk"
+	"google.golang.org/grpc/metadata"
+)
+
+// Broadcast is a proxy between EventBus and Kadcast service. It subscribes for
+// both topics.Kadcast and topics.KadcastPoint, compiles a valid wire frame and
+// propagates the message to Kadcast service.
+type Broadcast struct {
+	Base
+}
+
+// NewBroadcast ...
+func NewBroadcast(ctx context.Context, s eventbus.Subscriber, g *protocol.Gossip, rusk rusk.NetworkClient) ring.Writer {
+	b := &Broadcast{
+		Base: Base{
+			subscriber: s,
+			gossip:     g,
+			client:     rusk,
+			ctx:        ctx,
+			topic:      topics.Kadcast,
+		},
+	}
+
+	b.Subscribe()
+	return b
+}
+
+// Subscribe subscribes to eventbus Kadcast messages.
+func (w *Broadcast) Subscribe() {
+	mapper := func(t topics.Topic) byte {
+		const (
+			High = byte(1)
+			Low  = byte(0)
+		)
```

Comment:
Are these priority a new feature? If so a new tracking issue should be created to assess any potential attacking surface

Reply:
No, not a new feature. See also master: https://github.com/dusk-network/dusk-blockchain/blob/1d98aa4e1faa4fc1638a8fa8da0471e8c9274cc8/pkg/p2p/kadcast/writer.go#L71